packages/drivers/driver-web-cache/src/FluidCache.ts

```diff
@@ -117,10 +135,53 @@ export class FluidCache implements IPersistedCache {
 		});
 	}
 
+	private async openDb() {
+		if (this.noImmediateClose) {
+			if (this.db === undefined) {
+				const dbInstance = await getFluidCacheIndexedDbInstance(this.logger);
+				if (this.db === undefined) {
+					this.db = dbInstance;
+				} else {
+					dbInstance.close();
+					return this.db;
+				}
+				// Need to close the db on version change if opened.
+				this.db.onversionchange = (ev) => {
+					this.db?.close();
+					this.db = undefined;
+					clearTimeout(this.dbCloseTimer);
+					this.dbCloseTimer = undefined;
+				};
+				this.db.addEventListener("close", (ev) => {
+					clearTimeout(this.dbCloseTimer);
+					this.dbCloseTimer = undefined;
+					this.db = undefined;
+				});
+				// Schedule db close after this.closeDbAfter.
+				assert(this.dbCloseTimer === undefined, "timer should not be set yet!!");
```

Comment:
Out of curiosity what does `assert` do in consumer code? I don't see any other instances of this in `FluidCache.ts`

Reply:
It will throw exception in case it is false. We use it in FF code.

Reply:
In that case, my next question is whether this is so exceptional that the module is unusable and worth blowing up the caller over? Or is it something that can be observed over telemetry but should otherwise work fine?

Reply:
We don't blow up. All the db open are wrapped in try catches where we just record telemetry and go on.